src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -34,32 +36,40 @@ class TaskKiller @Inject() (
       case Some(app) =>
         checkAuthorization(UpdateRunSpec, app)
 
+        // TODO: We probably want to pass the execution context as an implcit.
         import scala.concurrent.ExecutionContext.Implicits.global
-        taskTracker.appTasks(appId).flatMap { allTasks =>
+
+        async {
```

Comment:
I talked to Jason about Scala Async and agreed that it has advantages over nested flatMaps or Scala's for comprehensions. E.g. it easier to reason about and avoids eird things like Future.sucessful(...) or the map.